gensim/models/phrases.py

```diff
@@ -425,6 +396,10 @@ def learn_vocab(sentences, max_vocab_size, delimiter=b'_', progress_per=10000,
                     "PROGRESS: at sentence #%i, processed %i words and %i word types",
                     sentence_no, total_words, len(vocab),
                 )
+            
+            if doc2vec:
+              sentence = sentence.words
```

Comment:
Probably here you "unapack" `TaggedDocument` class, you should do it before you pass it to phrases, no additional code needed here, something like

```python
sentences_for_d2v =  # list of TaggedDocument

sentences_for_phrases = [s.words for s in sentences]
phr = Phrases(sentences_for_phrases)

```